### ISSUE TITLE
DOM element attribute being set empty after component root object update

### ISSUE THREAD

**Issue by uakfdotb** [vuejs/vue#11538]
### Version
2.6.11

### Reproduction link
[https://jsfiddle.net/uxz2dnqv/](https://jsfiddle.net/uxz2dnqv/)

### Steps to reproduce
Using minimal reproduction: in the top select, change from 0 to 1. Then, inspect bottom select: the "Input 0" option should have the static attribute value="s0" as defined in the Vue.js template, but instead has value="".

`<option v-if="!specSet['s0']" value="s0">Input 0</option>`

### What is expected?
I thought Vue.js should refresh the UI after changing top select from 0 to 1 so that the Input 0 option appears, with value="s0".

`<option value="s0">Input 0</option>`

### What is actually happening?
Vue.js refreshes the UI and adds the Input 0 option, but its value attribute is set to empty string.

`<option value="">Input 0</option>`

---
It may be related to #11398

I have a parent component X with nested attribute X.Y.Z, and a child component that is given property referencing X.Y.Z.

The child component has a select input with options that appear depending on X.Y.Z.

When X.Y is updated (after changing the top select in the minimal reproduction), the options in the select input are updated but their attributes are not updated correctly.

**Comment by posva** [comment]
Make sure to `key` the elements in the loop: 

```html
<template v-for="node in query.Nodes">
  <option :key="node.ID" v-if="!specSet['n' + node.ID]" :value="'n' + node.ID">{{ node.ID }}</option>
</template>
```

Please when reporting a bug provide a straight forward reproduction, this one is very convoluted and hard to follow

**Comment by uakfdotb** [comment]
Thanks! Sorry for the incorrect bug, I missed this part of the docs.
